flang/include/flang/Lower/SymbolMap.h

```diff
@@ -295,6 +295,12 @@ class SymMap {
     return lookupSymbol(*sym);
   }
 
+  /// Find `symbol` and return its value if it appears in the top level map.
+  SymbolBox shallowLookupSymbol(semantics::SymbolRef sym);
```

Comment:
Could this be done with a `bool` argument to the current`lookupSymbol` calls, something like `lookupSymbol(semantics::SymbolRef sym, bool innermostScopeOnly = false)`?

Reply:
Yes, it clearly could. I'm not in favor of adding an extra argument to overload the function name. In the end, the one search has a loop and the other does not. I don't see a reason to merge the two into a single function.

Reply:
I also favor having different entry points like Eric rather than having a bool argument .